lib/python/rose/suite_engine_procs/cylc.py

```diff
@@ -600,6 +600,8 @@ def _parse_user_host(self, auth=None, user=None, host=None):
             host = auth
             if "@" in auth:
                 user, host = auth.split("@", 1)
+        if isinstance(user, bytes):
+            user, host = user.decode(), host.decode()
```

Comment:
Are we certain that ``user`` & ``host`` have consistent types? After all, they may be generated from different (modules &) methods if both start out as ``None``.

Reply:
Nope - that's not safe. I'll use a for loop/comprehension.

Reply:
I think I have now actually pushed a response to this. D'oh!